app/com/linkedin/drelephant/tuning/engine/SparkHBTParamRecommender.java

```diff
@@ -49,9 +52,18 @@
   private int lastRunExecutorCore;
   private long lastRunExecutorMemoryOverhead;
   private long lastRunDriverMemoryOverhead;
+  private Integer lastRunExecutorInstances;
```

Comment:
Can't we use primitive int here? We can use -1 as the initial value to check if lastRun values have been set or not. Reason being, using class wrappers for primitive types leads to unnecessary boxing/unboxing.

Reply:
That's fine. I don't think it matter that much in this case.

Reply:
As a rule of thumb, primitives should be preferred unless wrapper classes are necessary. There are various disadvantages ranging from unnecessary object creation, heavy objects, etc. We would never know how the code would evolve hence typically it is discouraged.

I do agree though it may not cause a huge issue here but it is a good practice to follow. No need to update the PR for just this though.